compose/utils.py

```diff
@@ -26,25 +23,33 @@ def parallel_execute(command, containers, doing_msg, done_msg, **options):
     for container in containers:
         write_out_msg(stream, lines, container.name, doing_msg)
 
+    q = Queue()
+
     def container_command_execute(container, command, **options):
         try:
             getattr(container, command)(**options)
         except APIError as e:
             errors[container.name] = e.explanation
+        q.put(container)
+
+    for container in containers:
+        t = Thread(
+            target=container_command_execute,
+            args=(container, command),
+            kwargs=options,
+        )
+        t.daemon = True
+        t.start()
+
+    done = 0
 
-    with concurrent.futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
-        future_container = {
-            executor.submit(
-                container_command_execute,
-                container,
-                command,
-                **options
-            ): container for container in containers
-        }
-
-        for future in concurrent.futures.as_completed(future_container):
-            container = future_container[future]
```

Comment:
I'm curious if something like this would work

```python

try:
    for future in concurrent.futures.as_completed(future_container):		
        container = future_container[future]
except KeyboardInterupt:
    for future in future_container.values():
        future.cancel()

```